### PR TITLE
Fix LTO support

### PR DESCRIPTION
**Pull request by jgunthorpe**
Now that gcc has defined attribute symver we can make LTO support work on compilers that support the new attribute.

@nmorey @Honggang-LI

### CODE REVIEW

File: buildlib/FindLDSymVer.cmake [hunk 1]
```diff
@@ -27,15 +27,25 @@ else()
...
-check_c_source_compiles("
-void ibv_get_device_list_1(void);
-void ibv_get_device_list_1(void){}
-asm(\".symver ibv_get_device_list_1, ibv_get_device_list@IBVERBS_1.1\");
-void ibv_get_device_list_0(void);
-void ibv_get_device_list_0(void){}
-asm(\".symver ibv_get_device_list_0, ibv_get_device_list@@IBVERBS_1.0\");
-
-int main(int argc,const char *argv[]){return 0;}" _LDSYMVER_SUCCESS)
+if (HAVE_FUNC_ATTRIBUTE_SYMVER)
```
Comment (rleon) [inline buildlib/FindLDSymVer.cmake:30]: Why do we need if-else here?
Isn't HAVE_FUNC_ATTRIBUTE_SYMVER already checked that symver work?

**Comment by Honggang-LI** [comment]
"linux-rdma.rdma-core (Build Test Build RPMs for fedora31)" is build task for fedora33. As I wrote in https://github.com/linux-rdma/rdma-core/pull/879 , please update buildlib/azure-pipelines.yml .

I confirmed the "symver" attribute usage with epel7, epel8, fedora31, fedoar32, fedora33 and fedora34/rawhide. It works as expected.

The PR looks good for me. Thanks

File: buildlib/FindLDSymVer.cmake (hunk 1 above)
Comment (jgunthorpe) [reply buildlib/FindLDSymVer.cmake:30]: ATTRIBUTE_SYMVER is checking if the C compiler attribute is supported, this is testing if the linker option works. We need to use the attribute version or this test fails on LTO linkers.

**Comment by nmorey** [comment]
@jgunthorpe Seems to work well on Tumbleweed

**Pull request merged**